src/mozanalysis/experiment.py

```diff
@@ -129,11 +137,16 @@ class Experiment:
             before UTC midnight.
     """
 
-    experiment_slug = attr.ib()
+    experiment_slug = attr.ib(type=str, validator=attr.validators.instance_of(str))
     start_date = attr.ib()
     num_dates_enrollment = attr.ib(default=None)
     app_id = attr.ib(default=None)
     app_name = attr.ib(default=None)
+    analysis_unit = attr.ib(
+        type=AnalysisUnit,
+        default=AnalysisUnit.CLIENT,
+        converter=lambda s: AnalysisUnit(s),  # allows callers to pass a string
```

Comment:
Is it desirable to allow calls to pass a string?

Reply:
I thought it was back when `AnalysisUnit` lived in mozanalysis, but now that the object is defined externally, in metric-hub, I think it's probably safe to force callers to use the enum